src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -89,6 +89,12 @@ public OpenIdConnectHandler(HttpClient backchannel)
                     message.PostLogoutRedirectUri = Options.PostLogoutRedirectUri;
                 }
 
+                var ticket = await HandleAuthenticateOnceAsync();
+                if (ticket != null)
+                {
+                    message.IdTokenHint = ticket.Principal?.FindFirst(OpenIdConnectParameterNames.IdToken)?.Value;
```

Comment:
@leastprivilege I know you're massively using the `RedirectToIdentityProvider` event to set the `id_token_hint` in your samples. Any objection if I add it directly to the OIDC handler?